### PR TITLE
Numpy int64 - known issues - docs changed fixes #8527

### LINKED ISSUE #8527: numpy.int64 doesn't decompose input
I didn't see this in the known issues, but @abonaca and I noticed that with numpy 1.15.4 + astropy 3.1.2:

```python
>>> np.int64((15*u.km) / (15*u.imperial.foot))
1
>>> np.int((15*u.km) / (15*u.imperial.foot))
3280
```

Any thoughts @mhvk?

### PR DESCRIPTION
**Pull request by disha012**
Fix #8527

### CODE REVIEW

**Comment by pllim** [comment]
Please make sure you can build the doc locally without warnings before pushing out a follow-up commit. Thanks!
```
/home/circleci/project/docs/known_issues.rst:220:py:obj reference target not found: np.int_
```

File: docs/known_issues.rst [hunk 1]
```diff
@@ -203,6 +203,23 @@ Colored printing of log messages and other colored text does work in Windows
...
+    1
+    >>> np.int((15 * u.km) / (15 * u.imperial.foot))
+    3280
+
+    >>> int is np.int
+    True
+    >>> np.int_ is np.int64
+    True
+
+Python's int or np.int goes through ``__index__`` while np.int64 or `np.int_` does not go through ``__index__``
```
Comment (mhvk) [inline docs/known_issues.rst]: Looks good modulo the missed links. Try `numpy.int64` and `numpy.int_`

Also, full stop at the end of this sentence, and break it to <80 characters.

Comment (adrn) [inline docs/known_issues.rst]: Here is a suggested replacement:
```suggestion
Python's ``int()`` or `numpy.int` goes through ``__index__`` while `numpy.int64` 
or `numpy.int_` does not go through ``__index__``
```

File: docs/known_issues.rst [hunk 2]
```diff
@@ -203,6 +203,23 @@ Colored printing of log messages and other colored text does work in Windows
 but only when running in the IPython console.  Colors are not currently
 supported in the basic Python command-line interpreter on Windows.
 
+numpy.int64 does not decompose input
```
Comment (adrn) [inline docs/known_issues.rst]:
```suggestion
numpy.int64 does not decompose input ``Quantity`` objects
```

File: docs/known_issues.rst [hunk 3]
```diff
@@ -203,6 +203,23 @@ Colored printing of log messages and other colored text does work in Windows
...
+------------------------------------
```
Comment (adrn) [inline docs/known_issues.rst]:
```suggestion
---------------------------------------------------------
```

File: docs/known_issues.rst [hunk 4]
```diff
@@ -203,6 +203,23 @@ Colored printing of log messages and other colored text does work in Windows
...
+::
+
+    >>> np.int64((15 * u.km) / (15 * u.imperial.foot))
+    1
+    >>> np.int((15 * u.km) / (15 * u.imperial.foot))
+    3280
+
+    >>> int is np.int
```
Comment (adrn) [inline docs/known_issues.rst]: I would remove this line and the example below in favor of an example above showing what happens with `int((15 * u.km) / (15 * u.imperial.foot))`.

File: docs/known_issues.rst [hunk 5]
```diff
@@ -203,6 +203,23 @@ Colored printing of log messages and other colored text does work in Windows
...
+    1
+    >>> np.int((15 * u.km) / (15 * u.imperial.foot))
+    3280
+    >>> int((15 * u.km) / (15 * u.imperial.foot))
+    3280
+
+Python's ``int()`` or `numpy.int` goes through ``__index__`` while 
+`numpy.int64` or `numpy.int_` does not go through ``__index__``.
+
+Hence int(...) is to be used if an integer is needed.
```
Comment (adrn) [inline docs/known_issues.rst]:
```suggestion
Hence `int(...)` should be used if an integer is needed from a dimensionless ``Quantity``.
```

**Comment by adrn** [comment]
Thanks @disha012!

**Pull request merged**